graph/push_v2.go

```diff
@@ -141,6 +146,13 @@ func (p *v2Pusher) pushV2Tag(tag string) error {
 		dgst, err := p.graph.GetDigest(layer.ID)
 		switch err {
 		case nil:
+			if p.layersPushed[layer.ID] {
```

Comment:
@dmcgowan @stevvooe shouldn't this be unnecessary in the future, when we could just Stat the registry for that hash and be sure that hash references the image we want?